WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift

```diff
@@ -193,29 +193,28 @@ private extension SiteStatsInsightsViewModel {
 
     func createMostPopularStatsRows() -> [StatsTotalRowData] {
         guard let mostPopularStats = store.getAnnualAndMostPopularTime(),
-                let mostPopularWeekday = mostPopularStats.mostPopularDayOfWeek.weekday,
-                let mostPopularHour = mostPopularStats.mostPopularHour.hour,
-                mostPopularStats.mostPopularDayOfWeekPercentage > 0
-         else {
+            var mostPopularWeekday = mostPopularStats.mostPopularDayOfWeek.weekday,
+            let mostPopularHour = mostPopularStats.mostPopularHour.hour,
+            mostPopularStats.mostPopularDayOfWeekPercentage > 0
+            else {
                 return []
         }
 
         var calendar = Calendar.init(identifier: .gregorian)
         calendar.locale = Locale.autoupdatingCurrent
 
-        let dayString = calendar.standaloneWeekdaySymbols[mostPopularWeekday - 1]
+        // Back up mostPopularWeekday by 1 to get correct index for standaloneWeekdaySymbols.
+        mostPopularWeekday = mostPopularWeekday == 0 ? 6 : mostPopularWeekday - 1
```

Comment:
Instead of hardcoding 6 here, what about of using 'calendar.standaloneWeekdaySymbols.count -1'

Reply:
Sure. And done!